### PR TITLE
Animal-pose by Cao et al.

### PR DESCRIPTION
**Pull request by jin-s13**

### CODE REVIEW

File: mmpose/datasets/datasets/__init__.py [hunk 1]
```diff
@@ -33,5 +33,5 @@
     'Face300WDataset', 'FaceAFLWDataset', 'FaceWFLWDataset', 'FaceCOFWDataset',
     'Body3DH36MDataset', 'AnimalHorse10Dataset', 'AnimalMacaqueDataset',
     'AnimalFlyDataset', 'AnimalLocustDataset', 'AnimalZebraDataset',
-    'AnimalATRWDataset'
+    'AnimalATRWDataset', AnimalPoseDataset
```
Comment (innerlee) [inline mmpose/datasets/datasets/__init__.py:36]: What's the relationship between AnimalPoseDataset and AnimalZebraDataset?

Comment (innerlee) [reply mmpose/datasets/datasets/__init__.py:36]: We have many dataset classes now, maybe its time to trim them down, and moving from code to configs.
Namely,  if user want to support a new dataset, they do not have to cp/paste/modify a dataset python file, instead they tweak a config file and voilà, works done.

**Review by innerlee — APPROVED** [review]

File: mmpose/datasets/datasets/animal/animal_pose_dataset.py [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+        self.ann_info['use_different_joint_weights'] = False
+        self.ann_info['joint_weights'] = np.array(
+            [
+                1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.2, 1.2, 1.2,
+                1.2, 1.5, 1.5, 1.5, 1.5
+            ],
+            dtype=np.float32).reshape((self.ann_info['num_joints'], 1))
+
+        # Note: The original paper did not provide enough information about
+        # the sigmas. We modified from 'https://github.com/cocodataset/'
```
Comment (innerlee) [inline mmpose/datasets/datasets/animal/animal_pose_dataset.py:98]: Can sigmas be computed from data?

Comment (jin-s13) [reply mmpose/datasets/datasets/animal/animal_pose_dataset.py:98]: No. The sigmas measure the accuracy of the annotation, which should be provided by the dataset.

Comment (innerlee) [reply mmpose/datasets/datasets/animal/animal_pose_dataset.py:98]: Okay

**Pull request merged**